syntax = "proto3";
package Proto;

import "Exchange.proto";
import "Request.proto";
import "Reply.proto";

message Holiday
{
  string date = 1;
  double weight = 2;
}

message TradingSession
{
  string begin = 1;
  string end = 2;
}

message ExchangeParameter
{
  Exchange exchange = 1;
  repeated TradingSession sessions = 2;
  repeated TradingSession maturity_sessions = 3;
  repeated Holiday holidays = 4;
  string charm_start_time = 5;
  int32 rfq_delay = 6;
  int32 rfq_timeout = 7;
  int32 rfq_volume = 8;
  string trading_day = 9;
  bool night_session = 10;
}

message ExchangeParameterReq
{
  RequestType type = 1;
  repeated ExchangeParameter parameters = 2;
  Exchange exchange = 3;
  string user = 4;
}

message ExchangeParameterRep
{
  repeated ExchangeParameter parameters = 1;
  Reply result = 2;
}
